src/main/typescript/components/form/parts/fileUpload/upload/FileLoader.tsx

```diff
@@ -148,16 +152,20 @@ const FileLoader = (props: FileLoaderProps) => {
         removeBeforeUpload()
     }
 
-    const shouldBlockNavigation = () => request !== undefined
+    const shouldBlockNavigation = () => requestRef.current !== undefined
```

Comment:
Not sure this works with a constant (don't know the meaning of `requestRef.current` either), as function you might need something like

```suggestion
    const shouldBlockNavigation = () => requestRef.current !== undefined && uploadStatus !== UploadStatus.PROCESSING
```

Reply:
I see what you mean. The `requestRef` is a 'holder' for the XHR. I had to change from using `useState` to `useRef` here (don't worry about the React hooks naming 😉) because `useState` causes an unwanted rerender whereas `useRef` doesn't.

The `requestRef` is set as soon as we have the XHR available and is only 'unset' (set to `undefined`) once it receives a response or is canceled. I would rather not allow the user to move away while the upload is processing. After all, this may go wrong, in which case we need to present the user with an error message.